### PR TITLE
Support stat a file through fileId

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes are proposed in this pull request?

Please outline the changes and how this PR fixes the issue.

### Why are the changes needed?

Please clarify why the changes are needed. For instance,
  1. If you propose a new API, clarify the use case for a new API.
  2. If you fix a bug, describe the bug.

### Does this PR introduce any user facing changes?

```Console

$ bin/alluxio fs stat -fileId 16844324863

/2 is a file path.

URIStatus{info=FileInfo{fileId=16844324863, name=2, path=/2, ufsPath=/Users/mbl/projects/github/alluxio-2/underFSStorage/2, length=4352, blockSizeBytes=67108864, creationTimeMs=1652321470886, completed=true, folder=false, pinned=false, pinnedlocation=[], cacheable=true, persisted=true, blockIds=[16827547648], inMemoryPercentage=0, lastModificationTimesMs=1648174994000, ttl=-1, lastAccessTimesMs=1648174994000, ttlAction=DELETE, owner=mbl, group=staff, mode=420, persistenceState=PERSISTED, mountPoint=false, replicationMax=-1, replicationMin=0, fileBlockInfos=[FileBlockInfo{blockInfo=BlockInfo{id=16827547648, length=4352, locations=[]}, offset=0, ufsLocations=[192.168.31.114]}], mountId=1, inAlluxioPercentage=0, ufsFingerprint=TYPE|FILE UFS|local OWNER|mbl GROUP|staff MODE|420 CONTENT_HASH|(len:4352,_modtime:1648174994000) , acl=user::rw-,group::r--,other::r--, defaultAcl=}, cacheContext=null}
Containing the following blocks: 
BlockInfo{id=16827547648, length=4352, locations=[]}
```

### CODE REVIEW

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java [hunk 1]
```diff
@@ -107,7 +122,7 @@ public String getUsage() {
 
   @Override
   public String getDescription() {
-    return "Displays info for the specified path both file and directory."
+    return "Displays info for the specified path or fileId both file and directory."
         + " Specify -f to display info in given format:"
         + "   \"%N\": name of the file;"
         + "   \"%z\": size of file in bytes;"
```
Comment (dbw9580) [inline shell/src/main/java/alluxio/cli/fs/command/StatCommand.java:128]: This section can use more formatting:

```suggestion
    return String.join("\n", Arrays.asList(
        "Displays info for the specified file or directory.",
        "Specify -fileId to treat the first positional argument as a file ID.",
        "Specify -f with a template string to display info in the given format. These format placeholders are available:",
        "  %N: name of the file",
        "  %z: size of file in bytes",
        ...
    ));
```

Comment (dbw9580) [reply shell/src/main/java/alluxio/cli/fs/command/StatCommand.java:128]: Adding a new format option `%i` to print the file Id if the user specifies the file with a path would be helpful. This way we can get a file's ID by its path.

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java [hunk 2]
```diff
@@ -94,7 +101,15 @@ protected void runPlainPath(AlluxioURI path, CommandLine cl)
   @Override
   public int run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
-    AlluxioURI path = new AlluxioURI(args[0]);
+    AlluxioURI path;
+    if (cl.hasOption("fileId")) {
+      try (CloseableResource<FileSystemMasterClient> client =
+          mFsContext.acquireMasterClientResource()) {
+        path = new AlluxioURI(client.get().getFilePath(Long.parseLong(args[0])));
```
Comment (dbw9580) [inline shell/src/main/java/alluxio/cli/fs/command/StatCommand.java:108]: Print a message that says `The specified file ID 123 is /path/to/file` here.

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java [hunk 3]
```diff
@@ -61,6 +63,11 @@ public Options getOptions() {
             .hasArg()
             .desc("format")
             .build()
+    ).addOption(
+        Option.builder("fileId")
```
Comment (dbw9580) [inline shell/src/main/java/alluxio/cli/fs/command/StatCommand.java:67]: Make it a `longOpt` `--file-id` to align with the convention in other commands?

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java [hunk 4]
```diff
@@ -61,6 +63,11 @@ public Options getOptions() {
...
+            .required(false)
+            .desc("fileId")
```
Comment (dbw9580) [inline shell/src/main/java/alluxio/cli/fs/command/StatCommand.java:69]: More detailed description for this new option?

**Review by dbw9580 — CHANGES_REQUESTED** [review]
Thanks for this improvement!

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java (hunk 2 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fs/command/StatCommand.java:108]: done

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java (hunk 4 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fs/command/StatCommand.java:69]: done

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java (hunk 3 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fs/command/StatCommand.java:67]: done

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java [hunk 5]
```diff
@@ -94,7 +103,17 @@ protected void runPlainPath(AlluxioURI path, CommandLine cl)
   @Override
   public int run(CommandLine cl) throws AlluxioException, IOException {
     String[] args = cl.getArgs();
-    AlluxioURI path = new AlluxioURI(args[0]);
+    AlluxioURI path;
+    if (cl.hasOption("file-id")) {
```
Comment (dbw9580) [inline shell/src/main/java/alluxio/cli/fs/command/StatCommand.java:107]: Make this string literal a static field `FILE_ID_OPTION` as in https://github.com/Alluxio/alluxio/blob/99e76a85141cdbdd7a96d13a6cc4a58c15728277/shell/src/main/java/alluxio/cli/fs/command/CpCommand.java#L90
and retrieve it by `FILE_ID_OPTION.getLongOpt()`.

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java [hunk 6]
```diff
@@ -107,16 +126,18 @@ public String getUsage() {
...
+        "Specify -file-id to treat the first positional argument as a file ID.",
+        "Specify -f to display info in given format:",
+        "   \"%N\": name of the file;",
+        "   \"%z\": size of file in bytes;",
+        "   \"%u\": owner;",
+        "   \"%g\": group name of owner;",
+        "   \"%y\" or \"%Y\": modification time,",
+        " %y shows 'yyyy-MM-dd HH:mm:ss' (the UTC date),",
+        " %Y it shows milliseconds since January 1, 1970 UTC;",
+        "   \"%b\": Number of blocks allocated for file"));
```
Comment (dbw9580) [inline shell/src/main/java/alluxio/cli/fs/command/StatCommand.java:140]: Can you please add a new format place holder `%i` that will be substituted with the file ID? This enables a user to get the file ID from its path. If you prefer, you can leave it to another PR.

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java [hunk 7]
```diff
@@ -107,16 +126,18 @@ public String getUsage() {
...
+        "Displays info for the specified file or directory.",
+        "Specify -file-id to treat the first positional argument as a file ID.",
+        "Specify -f to display info in given format:",
+        "   \"%N\": name of the file;",
+        "   \"%z\": size of file in bytes;",
+        "   \"%u\": owner;",
+        "   \"%g\": group name of owner;",
+        "   \"%y\" or \"%Y\": modification time,",
+        " %y shows 'yyyy-MM-dd HH:mm:ss' (the UTC date),",
+        " %Y it shows milliseconds since January 1, 1970 UTC;",
```
Comment (dbw9580) [inline shell/src/main/java/alluxio/cli/fs/command/StatCommand.java:139]:
```suggestion
        "   \"%y\": modification time in UTC in 'yyyy-MM-dd HH:mm:ss' format;",
        "   \"%Y\": modification time as Unix timestamp in milliseconds;",
```

**Review by dbw9580 — CHANGES_REQUESTED** [review]
Mostly LGTM with a few minor issues.

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java (hunk 6 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fs/command/StatCommand.java]: https://github.com/Alluxio/alluxio/issues/15642 I create an issue to track this.

File: shell/src/main/java/alluxio/cli/fs/command/StatCommand.java (hunk 5 above)
Comment (maobaolong) [reply shell/src/main/java/alluxio/cli/fs/command/StatCommand.java]: done

**Comment by maobaolong** [comment]
@dbw9580 Thanks for the previous review. PTAL.

**Review by dbw9580 — APPROVED** [review]
LGTM, thanks!

**Comment by maobaolong** [comment]
The failed test is DistributedCommandsStatsTest, it seems flaky test.

**Comment by dbw9580** [comment]
Error is known flaky test:

```
Error: 6.641 [ERROR] Errors: 
Error: 6.641 [ERROR] alluxio.client.cli.job.DistributedCommandsStatsTest.testAsyncPersistCancelStats
Error: 6.641 [ERROR]   Run 1: DistributedCommandsStatsTest.testAsyncPersistCancelStats:221 » Timeout Timed o...
Error: 6.642 [ERROR]   Run 2: DistributedCommandsStatsTest.testAsyncPersistCancelStats:221 » Timeout Timed o...
```

**Comment by HelloHorizon** [comment]
@maobaolong can you fix the conflict?

**Comment by maobaolong** [comment]
@HelloHorizon Done, PTAL

**Comment by HelloHorizon** [comment]
alluxio-bot, merge this please
